core/sugartypes.ml

```diff
@@ -566,6 +569,11 @@ let tappl' : phrase * tyarg list -> phrasenode = fun (e, tys) ->
        in
        TAppl (e, List.map make_arg tys)
 
+let get_normal_funlit fnlit =
```

Comment:
Ah -- I see you have done this already! Why not replace all the explicit unwrapping elsewhere with calls to this?

Reply:
Done. This one is newly added, there were some historical boilerplate code I forgot to check :p